src/jumpstart/saga.ts

```diff
@@ -0,0 +1,219 @@
+import BigNumber from 'bignumber.js'
+import walletJumpstart from 'src/abis/IWalletJumpstart'
+import { JumpstartEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { jumpstartLinkHandler } from 'src/jumpstart/jumpstartLinkHandler'
+import {
+  jumpstartClaimFailed,
+  jumpstartClaimStarted,
+  jumpstartClaimSucceeded,
+} from 'src/jumpstart/slice'
+import { NftMetadata } from 'src/nfts/types'
+import { getDynamicConfigParams } from 'src/statsig'
+import { DynamicConfigs } from 'src/statsig/constants'
+import { StatsigDynamicConfigs } from 'src/statsig/types'
+import { tokensByIdSelector } from 'src/tokens/selectors'
+import { getTokenId } from 'src/tokens/utils'
+import { addStandbyTransaction } from 'src/transactions/actions'
+import { NetworkId, TokenTransactionTypeV2 } from 'src/transactions/types'
+import Logger from 'src/utils/Logger'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import { publicClient } from 'src/viem'
+import { networkIdToNetwork } from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { all, call, fork, put, select } from 'typed-redux-saga'
+import { Hash, TransactionReceipt, parseAbi, parseEventLogs } from 'viem'
+
+const TAG = 'WalletJumpstart'
+
+export function* jumpstartClaim(privateKey: string) {
+  try {
+    yield* put(jumpstartClaimStarted())
+
+    const walletAddress = yield* select(walletAddressSelector)
+    if (!walletAddress) {
+      throw new Error('No wallet address found in store. This should never happen.')
+    }
+
+    const jumpstartContracts = getDynamicConfigParams(
+      DynamicConfigs[StatsigDynamicConfigs.WALLET_JUMPSTART_CONFIG]
+    ).jumpstartContracts
+
+    const networkIds = Object.keys(jumpstartContracts) as Array<keyof typeof jumpstartContracts>
+
+    let successfulClaimsCount = 0
+
+    for (const networkId of networkIds) {
+      try {
+        const contractAddress = jumpstartContracts[networkId]?.contractAddress
+
+        if (!contractAddress) {
+          throw new Error(
+            `Contract address for ${networkId} is not provided in dynamic config. This should never happen.`
+          )
+        }
+
+        const transactionHashes = yield* call(
+          jumpstartLinkHandler,
+          networkId,
+          contractAddress,
+          privateKey,
+          walletAddress
+        )
+
+        successfulClaimsCount += transactionHashes.length
+
+        yield* fork(dispatchPendingTransactions, networkId, transactionHashes)
+      } catch (error) {
+        Logger.error(TAG, `Error handling jumpstart link for ${networkId}`, error)
```

Comment:
this claim saga is kicked off every time a jumpstart deep link is consumed, but presumably each deep link is only for one network. if we had multichain jumpstart, this logger.error would always be fired because some chains will always have no transactions found right? and we'd always be doing unnecessary work, to try to claim across chains when we should only do it for one chain? perhaps what would be more efficient is to have the jumpstart link contain the network id...what do you think? sorry, i know this is different to what i suggested last time

Reply:
yes, I agree that we have to store the chain id somewhere in the jumpstart link to avoid unnecessary looping over all supported networks.
should we have a separate task for that?

in the meanwhile we can assume that we get network id from the link, right?

Reply:
yes let's do that! since we create the link when the user makes the transaction, we'll already have this information and no reason why we can't add it to the link to make our lives easier

Reply:
let's make the network id `pathParts[3]` of the jumpstart link